rr_platform/src/ultrasonic_array/ultrasonic_array_to_pcl.cpp

```diff
@@ -0,0 +1,232 @@
+#include <ros/ros.h>
+#include <ros/publisher.h>
+#include <sensor_msgs/PointCloud2.h>
+#include <pcl_conversions/pcl_conversions.h>
+#include <pcl_ros/transforms.h>
+#include <tf/transform_listener.h>
+#include <boost/asio.hpp>
+#include <boost/algorithm/string.hpp>
+#include <boost/math/constants/constants.hpp>
+
+using namespace std;
+
+/*
+  README
+  @author Brian Cochran, github: @btdubs
+  Node for taking in serial CSV distances from ultrasonic array,
+  then outputing a point cloud. Ultrasonic sensors defined in
+  urdf and uses that for transforming points.
+  Requires a tf tree being published.
+*/
+
+
+//defaults for launch file paramenters
+#define NUM_SENSORS_DEFAULT 6
+#define DRAW_METHOD_DEFAULT 0 //0 = Single Point, 1 = wall, 2 = circle, 3 = semicircle
+#define NUM_POINTS_DEFAULT 12 //# minimum points for each semicircle/circle/wall
+#define RADIUS_DEFAULT 0.1f //radius of semicircle/circle in meters
+#define LENGTH_DEFAULT 0.1f //length of wall
+#define DISTANCE_CLIP_DEFAULT 5.0f //distance beyond which we ignore point in meters
+#define BAUD_RATE_DEFAULT 9600 //rate of transfer. Needs to match Arduino
+
+const double pi = boost::math::constants::pi<double>();
+
+
+string readLine(boost::asio::serial_port &port) {
+    string line = "";
+    bool inLine = false;
+    while (true) {
+        char in;
+        try {
+            boost::asio::read(port, boost::asio::buffer(&in, 1));
+            if (in == '\n') {
+                return line;
+            }
+            line += in;
+        } catch (
+            boost::exception_detail::clone_impl <boost::exception_detail::error_info_injector<boost::system::system_error>> &err) {
+            ROS_ERROR("Error reading serial port.");
+            ROS_ERROR_STREAM(err.what());
+        }
+
+    }
+}
+
+vector<double> parseLine(string line) {
+  vector<double> dist;
+  vector<string> strs;
+  boost::split(strs, line, boost::is_any_of(","));
+
+  for (int i = 0; i < strs.size(); i++) {
```

Comment:
This is a good place to use [std::transform](http://en.cppreference.com/w/cpp/algorithm/transform)